finalPrep/modZeroHero.js

```diff
@@ -56,29 +94,74 @@ var excuse = 'I think I forgot to lock up my 1992 Toyota Coralla. Be right back.
 //   - luckyNumbers (array)
 //   - address (object with following key/values: number , street , state, zip)
 
+var scaryMonster = {
+  name: "Mothra",
+  smell: "*I Think You Should Leave Voice* Stinky!!!",
+  weight: 70000,
+  citiesDestroyed:["Sweatertown", "Coatsville", "Blanketopolis"],
+  luckyNumbers:[7,8,10],
+  address: {
+    number: 2710,
+    street: "O Street",
+    state: "California",
+    zip: "95816"
+  }
+}
 
 // Create a new class called SuperHero
 // - Your class should have the following DYNAMIC values
-//   - name 
+//   - name
 //   - superpower
-//   - age 
+//   - age
 // - Your class should have the following STATIC values
 //   - archNemesis, assigned to "The Syntax Error"
 //   - powerLevel = 100
-//   - energyLevel = 50 
+//   - energyLevel = 50
+
+class SuperHero {
+  constructor(name, superpower, age) {
+    this.name = name;
+    this.superpower = superpower;
+    this.age = age;
+    this.archNemesis = "The Syntax Error";
+    this.powerLevel = 100;
+    this.energyLevel = 50;
+  }
+  sayName() {
+    console.log(this.name);
+  }
+  maximizeEnergy() {
+    this.energyLevel = 1000
+  }
+  gainPower(newPower) {
+    if (typeof newPower === 'number') {
```

Comment:
I like the idea behind checking the type of input here. When it comes to classes, you can often skip that, because class methods will be used in very specific ways that the developer has control of, but this kind of mindset will serve you well with error handling.

Reply:
That makes sense, thank you! This piece of code was in response to the phrase "should take an argument of a number". I thought I had to ensure that it ONLY accepted numbers